scio-extra/src/main/scala/com/spotify/scio/extra/transforms/package.scala

```diff
@@ -158,4 +161,45 @@ package object transforms {
     }
   }
 
+  /**
+   * Enhanced version of [[SCollection]] with specialized versions of flatMap.
+   */
+  implicit class SpecializedFlatMapSCollection[T: ClassTag](val self: SCollection[T]) {
+
+    /**
+     * Latency optimized flavour of [[self.flatMap]], it returns a new SCollection by first
+     * applying a function to all elements of this SCollection, and then flattening the results.
+     * If function throws an exception, instead of retrying, faulty element goes into given error
+     * side output.
+     *
+     * @group transform
+     */
+    def safeFlatMap[U: ClassTag](f: T => TraversableOnce[U])
+    : (SCollection[U], SCollection[(T, Throwable)]) = {
+      val (mainTag, errorTag) = (new TupleTag[U], new TupleTag[(T, Throwable)])
+      val doFn = new NamedDoFn[T, U] {
+        val g = ClosureCleaner(f) // defeat closure
+        @ProcessElement
+        private[scio] def processElement(c: DoFn[T, U]#ProcessContext): Unit = {
+          val i = Try {
```

Comment:
use `try/catch` instead to avoid extra tiny objects?
I'll normally use `Try` as function return type when `match` `Success`/`Failure` happens in a separate place.